File: 1227-number-of-equivalent-domino-pairs.java
class Solution {
    public static int numEquivDominoPairs(int[][] dominoes) {
        int[] counter = new int[100]; 
        for (int[] domino : dominoes) {
            int a = domino[0], b = domino[1];
            int normalized = a < b ? a * 10 + b : b * 10 + a;
            counter[normalized]++;
        }
        int totalPairs = 0;
        for (int count : counter) {
            totalPairs += count * (count - 1) / 2;
        }
        return totalPairs;
    }
}

File: python/1227-number-of-equivalent-domino-pairs.py
class Solution:
    @staticmethod
    def numEquivDominoPairs(dominoes: List[List[int]]) -> int:
        counter = [0] * 100  
        for domino in dominoes:
            a, b = domino
            normalized = a * 10 + b if a < b else b * 10 + a
            counter[normalized] += 1
        return sum(count * (count - 1) // 2 for count in counter)
